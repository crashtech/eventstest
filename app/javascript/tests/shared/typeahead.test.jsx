import React from 'react'
import renderer from 'react-test-renderer'

let promise = null

jest.mock('@api')
import { TypeaheadApi } from '@api'
TypeaheadApi.mockReturnValue(() => (
  promise = Promise.resolve({ data: [{ value: 1, text: 'Option 1' }] })
))

import Typeahead from '../../shared/typeahead'

jest.useFakeTimers()
test('Can load options', () => {
  const component = renderer.create(<Typeahead source="test" />)
  expect(component.toJSON()).toMatchSnapshot()

  component.getInstance().loadOptions(null, { searchQuery: 'Query' })
  jest.runAllTimers()

  promise.then(() => {
    component.getInstance().forceUpdate()
    expect(component.toJSON()).toMatchSnapshot()
  })
})

test('Can add new options', () => {
  const component = renderer.create(<Typeahead source="test" />)
  expect(component.toJSON()).toMatchSnapshot()

  component.getInstance().addItem(null, { value: 'New Option' })
  component.getInstance().forceUpdate()
  expect(component.toJSON()).toMatchSnapshot()
})
