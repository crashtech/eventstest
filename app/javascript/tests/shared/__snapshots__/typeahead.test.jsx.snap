// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can add new options 1`] = `
<div
  aria-busy={false}
  aria-expanded={false}
  className="ui search selection dropdown"
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  role="combobox"
>
  <input
    aria-autocomplete="list"
    autoComplete="off"
    className="search"
    onChange={[Function]}
    style={
      Object {
        "width": undefined,
      }
    }
    tabIndex={0}
    type="text"
    value=""
  />
  <div
    aria-atomic={true}
    aria-live="polite"
    className="text"
    role="alert"
  />
  <i
    aria-hidden="true"
    className="dropdown icon"
    onClick={[Function]}
  />
  <div
    className="menu transition"
    role="listbox"
  >
    <div
      className="message"
    >
      No results found.
    </div>
  </div>
</div>
`;

exports[`Can add new options 2`] = `
<div
  aria-busy={false}
  aria-expanded={false}
  className="ui search selection dropdown"
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  role="combobox"
>
  <input
    aria-autocomplete="list"
    autoComplete="off"
    className="search"
    onChange={[Function]}
    style={
      Object {
        "width": undefined,
      }
    }
    tabIndex={0}
    type="text"
    value=""
  />
  <div
    aria-atomic={true}
    aria-live="polite"
    className="text"
    role="alert"
  />
  <i
    aria-hidden="true"
    className="dropdown icon"
    onClick={[Function]}
  />
  <div
    className="menu transition"
    role="listbox"
  >
    <div
      aria-checked={false}
      aria-selected={true}
      className="selected item"
      onClick={[Function]}
      role="option"
      style={
        Object {
          "pointerEvents": "all",
        }
      }
    >
      <span
        className="text"
      >
        New Option
      </span>
    </div>
  </div>
</div>
`;

exports[`Can load options 1`] = `
<div
  aria-busy={false}
  aria-expanded={false}
  className="ui search selection dropdown"
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  role="combobox"
>
  <input
    aria-autocomplete="list"
    autoComplete="off"
    className="search"
    onChange={[Function]}
    style={
      Object {
        "width": undefined,
      }
    }
    tabIndex={0}
    type="text"
    value=""
  />
  <div
    aria-atomic={true}
    aria-live="polite"
    className="text"
    role="alert"
  />
  <i
    aria-hidden="true"
    className="dropdown icon"
    onClick={[Function]}
  />
  <div
    className="menu transition"
    role="listbox"
  >
    <div
      className="message"
    >
      No results found.
    </div>
  </div>
</div>
`;

exports[`Can load options 2`] = `
<div
  aria-busy={false}
  aria-expanded={false}
  className="ui search selection dropdown"
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  role="combobox"
>
  <input
    aria-autocomplete="list"
    autoComplete="off"
    className="search"
    onChange={[Function]}
    style={
      Object {
        "width": undefined,
      }
    }
    tabIndex={0}
    type="text"
    value=""
  />
  <div
    aria-atomic={true}
    aria-live="polite"
    className="text"
    role="alert"
  />
  <i
    aria-hidden="true"
    className="dropdown icon"
    onClick={[Function]}
  />
  <div
    className="menu transition"
    role="listbox"
  >
    <div
      aria-checked={false}
      aria-selected={true}
      className="selected item"
      onClick={[Function]}
      role="option"
      style={
        Object {
          "pointerEvents": "all",
        }
      }
    >
      <span
        className="text"
      >
        Option 1
      </span>
    </div>
  </div>
</div>
`;
